build.gradle

```diff
@@ -47,9 +47,10 @@ plugins {
   // Updating the shadow plugin version to 8.1.1 causes issue with signing and publishing the shadowed
   // artifacts - see https://github.com/johnrengelman/shadow/issues/901
   id 'com.github.johnrengelman.shadow' version '8.1.0' apply false
-  //  the minimum required JRE of 6.14.0+ is 11
-  //  refer:https://github.com/diffplug/spotless/tree/main/plugin-gradle#requirements
-  id 'com.diffplug.spotless' version "6.13.0" apply false
+  //  Spotless 6.13.0 has issue with Java 21 (see https://github.com/diffplug/spotless/pull/1920), and Spotless 6.14.0+ requires JRE 11
+  //  We are going to drop JDK8 support. Hence, the spotless is upgrade to newest version and be applied only if the build env is compatible with JDK 11.
+  //  spotless 6.15.0+ has issue in runtime with JDK8 even through we define it with `apply:false`. see https://github.com/diffplug/spotless/issues/2156 for more details
+  id 'com.diffplug.spotless' version "6.14.0" apply false
```

Comment:
```suggestion
  id 'com.diffplug.spotless' version "6.25.0" apply false
```

Reply:
You can always use the latest Spotless and bump your JRE to match it.

Reply:
@Goooler thanks for your comment. 

Kafka still support JDK 8, so we make CI run checks/tests with JDK8. However, as @gongxuanzhang described, spotless 6.15+ can't work with JDK8 even though we set "apply false".

Reply:
yep, you are right. We have set the `sourceCompatibility=8`

https://github.com/apache/kafka/blob/trunk/build.gradle#L57
https://github.com/apache/kafka/blob/trunk/build.gradle#L318

I know that we can use JDK11 to build kafak which is compatible with JDK8. However, our rules is "we ought to run all checks/tests with all supported JDKs". That is why we need to make sure spotless can work with all supported JDKs.

for another, we will drop JDK8 in next release. Hence, we can remove this workaround and upgrade spotless to newest version later.

Reply:
Good to see.

Reply:
@Goooler Do you approve this PR? it would be great to get approve from spotless core developer :)

Reply:
Sorry for the delay.

I think it would be great to throw exceptions if the current JRE is not Java 11 compatible.

Reply:
> I think it would be great to throw exceptions if the current JRE is not Java 11 compatible.

I had similar idea before (see https://github.com/apache/kafka/pull/13311#discussion_r1118148218) . However, that means our developers can't use JDK8 to write code for kafka. Also, we can't follow our JDK rules - run all checks/tests with all supported JDKs